chia/data_layer/data_layer_wallet.py

```diff
@@ -209,6 +211,7 @@ async def new_launcher_spend_response(self, response: PuzzleSolutionResponse, ac
             CoinSpend(launcher_coin, response.puzzle, response.solution),
             height=response.height,
         )
+        await self.wallet_state_manager.action_store.action_done(action_id)
```

Comment:
I have no idea what is better, but apparently there is `WalletStateManager.set_action_done()` that passes through.

This feels like a thing where a context manager would be appropriate with explicit non-success behavior.  As is I guess we just build up the action table in the db without any culling of no-longer-needed actions, along with completed actions.  It also seems like the action update should be in a transaction with the rest of the db change associated with it?

Presumably this is either just me needing to learn more about this, or if a change is appropriate then at least something to be addressed in separate PR.